scene/resources/visual_shader_particle_nodes.cpp

```diff
@@ -504,66 +502,88 @@ void VisualShaderNodeParticleMeshEmitter::_update_textures() {
 	Vector<Vector2> uvs2;
 
 	if (use_all_surfaces) {
-		for (int i = 0; i < max_surface_index; i++) {
+		for (int i = 0; i < mesh->get_surface_count(); i++) {
 			// position
-			Array vertex_array = mesh->surface_get_arrays(i)[Mesh::ARRAY_VERTEX];
-			for (int j = 0; j < vertex_array.size(); j++) {
-				vertices.push_back((Vector3)vertex_array[j]);
+			if (mesh->surface_get_arrays(i).size() > Mesh::ARRAY_VERTEX) {
+				Array vertex_array = mesh->surface_get_arrays(i)[Mesh::ARRAY_VERTEX];
```

Comment:
Sounds like both `mesh->surface_get_arrays(i)` and `mesh->surface_get_arrays(i).size()` could be saved to local variables to avoid calling the getters for every check:
```
const Array surface_arrays = mesh->surface_get_arrays(i);
const int surface_arrays_size = surface_arrays.size();
```

Reply:
Done